### PR TITLE
Feature/robot libdoc enhancements

### PR DESCRIPTION
**Pull request by boakley**
This change lets you use wildcards to define the path option of the robot_libdoc task. There's also a bugfix that catches errors when creating the output, where it throws a proper `TaskOptionsError` rather than just spew a traceback. 

This includes a new function in utils, `process_glob_list_arg` which works like `process_list_arg`, but goes the extra step and expands wildcards.

# Critical Changes

# Changes

# Issues Closed

### CODE REVIEW

File: cumulusci/core/tests/test_utils.py [hunk 1]
```diff
@@ -34,6 +36,43 @@ def test_process_list_arg(self):
...
+            self.assertEqual(
+                ["*.bar", "x.y.z"], utils.process_glob_list_arg("*.bar, x.y.z")
+            )
+
+            # Recursive
+            os.mkdir("subdir")
+            touch("subdir/baz.resource")
+            self.assertEqual(
+                ["subdir/baz.resource"], utils.process_glob_list_arg("**/*.resource")
+            )
```
Comment (davisagli) [inline cumulusci/core/tests/test_utils.py]: This test is failing in Windows due to the different path separator. It looks like the glob does the right thing but you need to use os.path.join to build the expected result.

File: cumulusci/core/utils.py [hunk 2]
```diff
@@ -44,6 +45,31 @@ def process_bool_arg(arg):
...
+        return []
+
+    files = []
+    for path in initial_list:
+        more_files = glob.glob(path, recursive=True)
+        if len(more_files):
+            files += sorted(more_files)
+        else:
+            files.append(path)
+    return files
```
Comment (davisagli) [inline cumulusci/core/utils.py]: It's possible for multiple different globs to include the same file. So I think it'd be a good idea to work with `files` as a set and then convert to a list as you return it.

Comment (boakley) [reply cumulusci/core/utils.py]: hmmm. I see your point. Sets are unordered though, so we'll lose the ordering. That's not necessarily a bad thing, but it's a regression.

Comment (davisagli) [reply cumulusci/core/utils.py]: If ordering is important you can use a dict, which preserve insertion order in Python 3.6+

Comment (boakley) [reply cumulusci/core/utils.py]: yeah, that's what I did.

**Pull request merged**